pkg/prometheus/promcfg.go

```diff
@@ -2979,13 +2979,22 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	// DNSSDConfig
 	if len(sc.Spec.DNSSDConfigs) > 0 {
 		configs := make([][]yaml.MapItem, len(sc.Spec.DNSSDConfigs))
+
+		// A version check for Record Types A, AAAA and SRV is not necessary, but they were added to
+		// not break the code at `compatibilityMatrix[*config.Type]` in case those types occur.
+		compatibilityMatrix := map[monitoringv1alpha1.DNSRecordType]string{
+			monitoringv1alpha1.DNSRecordTypeA:    "0.16.0",
+			monitoringv1alpha1.DNSRecordTypeAAAA: "0.16.0",
+			monitoringv1alpha1.DNSRecordTypeSRV:  "0.16.0",
```

Comment:
since the operator requires Prometheus 2.x, it's a bit confusing to have "0.16.0" here (+ there's a risk of missing a supported value). See below for an alternative.